File: src/dev/wirlie/bungeecord/glist/config/ConfigEntry.java
package dev.wirlie.bungeecord.glist.config;

import dev.wirlie.bungeecord.glist.EnhancedBCL;
import org.jetbrains.annotations.NotNull;

public class ConfigEntry<T> {

    private final String key;

    private T value = null;

    public ConfigEntry(String key) {
        this.key = key;
        EnhancedBCL.CONFIGURATIONS_REGISTRY.add(this);
    }

    public void setValue(@NotNull Object value) {
        this.value = (T) value;
    }

    @NotNull
    public T get() {
        return value;
    }

    public String getKey() {
        return key;
    }

}

File: Platform-BungeeCord/src/main/kotlin/dev/wirlie/bungeecord/glist/config/ConfigEntry.kt
package dev.wirlie.bungeecord.glist.config

import dev.wirlie.bungeecord.glist.EnhancedBCL

class ConfigEntry<T>(val key: String) {
    private var value: T? = null

    init {
        EnhancedBCL.CONFIGURATIONS_REGISTRY.add(this)
    }

    fun setValue(value: Any) {
        @Suppress("UNCHECKED_CAST")
        this.value = value as T
    }

    fun get(): T {
        @Suppress("UNCHECKED_CAST")
        return value as T
    }
}
